src/main/ArrayManager.java

```diff
@@ -160,11 +181,11 @@ public void shuffleArray(int[] array, int currentLen, ArrayVisualizer ArrayVisua
             Delays.setSleepRatio(sleepRatio);
         }
 
-        Shuffles tempShuffle = this.Shuffles;
-        if(Distributions == Distributions.RANDOM || Distributions == Distributions.EQUAL)
-            this.Shuffles = Shuffles.ALREADY;
-        Shuffles.shuffleArray(array, this.ArrayVisualizer, Delays, Highlights, Writes);
-        this.Shuffles = tempShuffle;
+        ShuffleGraph tempShuffle = this.shuffle;
+        if(distribution == Distributions.RANDOM || distribution == Distributions.EQUAL)
+            this.shuffle = ShuffleGraph.single(Shuffles.ALREADY);
+        shuffle.shuffleArray(array, this.ArrayVisualizer);
```

Comment:
maybe scale the sleep ratio here based on how many shuffles there are?

Reply:
I don't agree

Reply:
if you want to make a complicated distribution, it'll be a lot slower (and shuffles/distributions are usually relatively simple)

Reply:
Maybe adding an option to control the speed of the shuffle manually, even maybe in the advanced shuffle editor, would be good.

Reply:
Done!